### ISSUE TITLE
port moin2 code to py3

### ISSUE THREAD

**Issue by ThomasWaldmann** [moinwiki/moin#720]
See there: https://github.com/moinwiki/moin/projects/1

**Comment by ThomasWaldmann** [comment]
If you want to help with this, something that can be done at any time is to improve testing:
- better test coverage
- better tests
- see also other testing-related tickets

Before we can port to py3 at all, we need to have good testing, otherwise we might introduce lots of issues and manual testing would slow down porting also.

**Comment by ThomasWaldmann** [comment]
I talked to some people at europython 2019 and changed my mind.

We rather port to py3 now:
- people can write new code in py3, which is more enjoyable and what modern developers might expect
- project looks less "dead"
- port to py3 **only** (not py2 + py3)

I suggest to require py35+ for now (as debian stable still had that before buster was released a few days ago) and later increase to py36+

**Comment by ReimarBauer** [comment]
We had some issues with our mss software with py35 and we are now based on py37. That does not mean that these problems will occure in moin too. I only remember that this was a problem on some of the platforms we support. 

We did the same step this year before gsoc starts and that made a big difference for all who want to contribute.

Thx for these good news.

**Comment by ThomasWaldmann** [comment]
We are going to py35+ only, no py2 compat needed, so we do not want to use `six`.

TODO: Fork `ThomasWaldmann:moin`, start with branch py3, create feature branch, pick one todo from below, carefully globally fix all related to it, commit like `py3: dict`, PR against py3 branch.

- [ ] 2to3 dict
- [ ] 2to3 xrange

- [ ] 2to3 imports
- [ ] 2to3 urllib

- [ ] python-modernize print

- [ ] 2to3 basestring
- [ ] 2to3 unicode
- [ ] python-modernize unichr: uses six.unichr

**Comment by ThomasWaldmann** [comment]
easy task (pep8 fixing): https://github.com/moinwiki/moin/pull/845#issuecomment-511183252

**Comment by ThomasWaldmann** [comment]
another task: https://github.com/moinwiki/moin/pull/845#issuecomment-511192283

I'll be away for a while, trying an ebike in basel area, so this is the end to ep2019 sprints for me now. :)

Any work on these recent issues (also see comment above) would be appreciated.

Please base your work on my PR code: https://github.com/moinwiki/moin/pull/845

**Comment by ThomasWaldmann** [comment]
Done and merged into master.

So moin2 is now a python3-only project. \o/
